packages/plugin-core/src/commands/MoveHeader.ts

```diff
@@ -0,0 +1,311 @@
+            resolve(undefined);
+          }
+        },
+      });
+    });
+  }
+
+  removePosition = (nodes: Node[]) => {
+    return _.map(nodes, (node) => {
+      visit(node, (n) => {
+        delete n["position"];
+      });
+      return node;
+    });
+  };
+
+  async execute(opts: CommandOpts): Promise<CommandOutput> {
+    const ctx = "MoveHeaderCommand";
+    this.L.info({ ctx, opts });
+    const {
+      dest,
+      origin,
+      nodesToMove,
+      modifiedOriginTree,
+      originProc,
+      engine,
+    } = opts;
+
+    // deep copy the origin before mutating it
+    const originDeepCopy = _.cloneDeep(origin);
+    // remove header from origin
+    const modifiedOriginContent = originProc.stringify(modifiedOriginTree);
+
+    origin.body = modifiedOriginContent;
+    await engine.writeNote(origin, {
+      updateExisting: true,
+    });
+
+    // append header to destination
+    const destProc = this.getProc(engine, dest!);
+    const destTree = destProc.parse(dest!.body);
+    destTree.children = (destTree.children as Node[]).concat(nodesToMove);
+    const modifiedDestContent = destProc.stringify(destTree);
+    dest!.body = modifiedDestContent;
+    await engine.writeNote(dest!, {
+      updateExisting: true,
+    });
+
+    delayedUpdateDecorations();
+
+    // update reference
+    const anchorsBefore = RemarkUtils.findAnchors(originDeepCopy.body);
+    const anchorsAfter = RemarkUtils.findAnchors(modifiedOriginContent);
+    const anchorsToUpdate = _.differenceWith(
+      anchorsBefore,
+      anchorsAfter,
+      this.hasIdenticalChildren
+    );
+    const anchorsToUpdateNames = _.map(anchorsToUpdate, (anchor) => {
+      const slugger = getSlugger();
+      const payload = AnchorUtils.anchorNode2anchor(anchor, slugger);
+      return payload![0];
+    });
+
+    const foundReferences = await findReferences(origin.fname);
+    const { vaults, wsRoot, notes } = engine;
+    const updated: NoteProps[] = [];
+    _.forEach(foundReferences, async (reference) => {
+      const { location, isCandidate } = reference;
+      // ignore candidate refs.
+      if (isCandidate) return false;
+      // get note from location
+      const fsPath = location.uri.fsPath;
+      const fname = NoteUtils.normalizeFname(path.basename(fsPath));
+      const vault = VaultUtils.getVaultByNotePath({
+        fsPath,
+        wsRoot,
+        vaults,
+      });
+      const note = NoteUtils.getNoteByFnameV5({
+        fname,
+        notes,
+        vault,
+        wsRoot,
+      });
+      if (!note) return false;
+
+      // find match text in note and modify the node
+      const proc = this.getProc(engine, note);
+      const tree = proc.parse(note.body);
+      visit(tree, (node: Node) => {
+        if (node.type === DendronASTTypes.WIKI_LINK) {
+          if (
+            node.value === origin.fname &&
+            _.includes(anchorsToUpdateNames, node.data!.anchorHeader)
+          ) {
+            node.value = dest!.fname;
```

Comment:
lets have a followup task to make it possible to move xvault. we can add it to `triage`